x-pack/osquerybeat/beater/config_plugin_test.go

```diff
@@ -116,13 +117,45 @@ func TestFlattenECSMapping(t *testing.T) {
 	}
 }
 
-func TestFlattenECSMappingTooDeep(t *testing.T) {
-	const mapping = `{"a":{"b":{"c":{"d":{"e":{"f":{"g":{"h":{"i":{"j":{"k":{"value": 1}}}}}}}}}}}}`
-	var m map[string]interface{}
-	err := json.Unmarshal([]byte(mapping), &m)
+func generateTestMapping(depth int, k string, v interface{}) map[string]interface{} {
+	m := make(map[string]interface{})
+	res := m
+	key := 'a'
+
+	for i := 0; i < depth; i++ {
+		newmap := make(map[string]interface{})
+		m[string(key)] = newmap
+		m = newmap
+		key += 1
+	}
+	m[k] = v
+	return res
+}
+
+func TestFlattenECSMappingEdges(t *testing.T) {
```

Comment:
do we also need a test for the error case?

Reply:
this test covers some error cases, will do another pass to double check what other cases can be added